src/gossip/service.zig

```diff
@@ -280,61 +284,64 @@ pub const GossipService = struct {
         // const echo_server_thread = try self.echo_server.listenAndServe();
         // _ = echo_server_thread;
 
-        var receiver_handle = try Thread.spawn(.{}, socket_utils.readSocket, .{
+        const receiver_handle = try Thread.spawn(.{}, socket_utils.readSocket, .{
             self.allocator,
             &self.gossip_socket,
             self.packet_incoming_channel,
             self.exit,
             self.logger,
         });
-        defer self.joinAndExit(&receiver_handle);
+        defer self.joinAndExit(receiver_handle);
 
-        var packet_verifier_handle = try Thread.spawn(.{}, verifyPackets, .{self});
-        defer self.joinAndExit(&packet_verifier_handle);
+        // TODO: use better allocator, unless GPA becomes more performant.
+        var gp_task_allocator: std.heap.GeneralPurposeAllocator(.{ .thread_safe = !builtin.single_threaded }) = .{};
+        defer _ = gp_task_allocator.deinit();
 
-        var packet_handle = try Thread.spawn(.{}, processMessages, .{self});
-        defer self.joinAndExit(&packet_handle);
+        var heap_task_allocator = if (builtin.os.tag == .windows) std.heap.HeapAllocator.init() else {};
+        defer if (builtin.os.tag == .windows) heap_task_allocator.deinit();
 
-        var maybe_build_messages_handle = if (!spy_node) try Thread.spawn(.{}, buildMessages, .{self}) else null;
-        defer {
-            if (maybe_build_messages_handle) |*handle| {
-                self.joinAndExit(handle);
-            }
-        }
+        const task_allocator = switch (builtin.os.tag) {
+            .windows => heap_task_allocator.allocator(),
+            else => gp_task_allocator.allocator(),
```

Comment:
in processesMessages loop we use `self.allocator` to shallow free - so using a different allocator here would break that rn (which you can see if u try to run gossip) - fyi should run most of the commands before opening pr (im not great at this too - ci/cd which runs gossip for a few seconds and then shuts down is on the todo list)